ec2/sc-ec2-linux-docker-notebook.yaml

```diff
@@ -0,0 +1,513 @@
+                NOTEBOOK_CONTAINER_NAME:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, DockerNotebookContainerName]
+                SERVICE_CATALOG_PREFIX: service-catalog/synapse/cred
+                SSM_PARAMETER_SUFFIX: oidc-accesstoken
+                NOTEBOOK_TYPE:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, NoteBookTypeForDocker]
+        make_work_dir:
+          commands:
+            make_work_dir:
+              command: |
+                mkdir -p ${EC2_WORK_DIR}
+                sudo chmod 777 ${EC2_WORK_DIR}
+              env:
+                EC2_WORK_DIR:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, EC2WorkDir]
+        start_jupyter_notebook:
+          commands:
+            start_notebook:
+              command: "docker start ${NOTEBOOK_CONTAINER_NAME} || \
+                docker run -d --name ${NOTEBOOK_CONTAINER_NAME} \
+                --restart unless-stopped \
+                -e DOCKER_STACKS_JUPYTER_CMD=notebook \
+                -v ${EC2_WORK_DIR}:${JUPYTER_WORK_DIR} \
+                --network ${NETWORK_NAME} ${DOCKER_IMAGE} \
+                start-notebook.py --IdentityProvider.token='' --NotebookApp.base_url=/$(curl http://169.254.169.254/latest/meta-data/instance-id) \
+                --notebook-dir=${JUPYTER_WORK_DIR}"
+              env:
+                NETWORK_NAME:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, DockerNetworkName]
+                NOTEBOOK_CONTAINER_NAME:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, DockerNotebookContainerName]
+                EC2_WORK_DIR:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, EC2WorkDir]
+                DOCKER_IMAGE:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, DockerImage]
+                JUPYTER_WORK_DIR: /home/jovyan/workdir
+        start_rstudio_notebook:
+          commands:
+            start_notebook:
+              command: "docker start ${NOTEBOOK_CONTAINER_NAME} || \
+                docker run -d --name ${NOTEBOOK_CONTAINER_NAME} \
+                --restart unless-stopped \
+                --network ${NETWORK_NAME} \
+                -e DISABLE_AUTH=true \
+                -v ${EC2_WORK_DIR}:/home/rstudio \
+                ${DOCKER_IMAGE}"
+              env:
+                NETWORK_NAME:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, DockerNetworkName]
+                NOTEBOOK_CONTAINER_NAME:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, DockerNotebookContainerName]
+                DOCKER_IMAGE:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, DockerImage]
+                EC2_WORK_DIR:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, EC2WorkDir]
```

Comment:
there's a `JUPYTER_WORK_DIR` env var `start_jupyter_notebook` but not one here?  for consistency maybe create a generic one called `APP_WORK_DIR` for both?